crawl-ref/source/initfile.cc

```diff
@@ -4024,6 +4023,14 @@ void get_system_environment()
         SysEnv.crawl_dir = SAVE_DIR_PATH;
 #endif
 
+#ifdef __ANDROID__
+    if (SysEnv.crawl_dir.empty())
+    {
+        SysEnv.crawl_dir = SDL_AndroidGetExternalStoragePath();
```

Comment:
Hi, @robertxgray , I'm working for CDDA and trying to upgrade its android build to adapt Android 11 https://github.com/CleverRaven/Cataclysm-DDA/issues/49310#L528

Could you please teach me how to make game save to the /media folder? AFAIK, this SDL_AndroidGetExternalStoragePath will return something like `/sdcard/Android/data/org.develz.crawl/saves/`, while Android 11 require that saves should be inside `/media/org.develz.crawl/saves/` so player can backup/modify/sync their save files easily.

Reply:
Hi again!

This is what I found in the documentation.

The DCSS app includes the WRITE_EXTERNAL_STORAGE permission in the manifest, but [it's no longer needed](https://developer.android.com/guide/topics/manifest/uses-permission-element) for the app to write it's own application-specific directories since Android 4.4.

This permission can be used to write in other directories, but [it must be explicitly granted by the user](https://developer.android.com/training/permissions/requesting) on devices running Android 6.0 or higher. I made some successful tests, for example using /storage/emulated/0/dcss instead of the path returned by SDL_AndroidGetExternalStoragePath.

The application-specific directories cannot be browsed on Android 11 using the stock file manager, but it can be done with other file managers like [Speed Software's Explorer](https://play.google.com/store/apps/details?id=com.speedsoftware.explorer).

I didn't find anything about /media/org.develz.crawl/saves, unless you mean the [MediaStore API](https://developer.android.com/training/data-storage/shared/media) which can be used to write on media paths without extra permissions. I'm not sure if it's a viable solution because it's designed for media files and I couldn't find an API for the native interface.

I think the best solution would be using the app-specific directory by default but giving the users an option to use another path instead. Only if they choose another path they'll need to grant extra permissions to the app. I prefer not requesting extra permissions unless strictly necessary.

But I don't plan to include this feature in DCSS for now, I don't even know where to include the option because the game doesn't have any configuration menus. I hope not many users want to access the files while using Android 11 and they can still do it by installing another file manager. What I'm going to do, is restricting the WRITE_EXTERNAL_STORAGE permission to Android 4.3 or lower in the next update.

Kind regards.